controllers/subscription_controller.go

```diff
@@ -117,6 +132,58 @@ func (r *SubscriptionReconciler) ensureSubscriptions(logger logr.Logger, namespa
 	return err
 }
 
+func (r *SubscriptionReconciler) setOperatorCondition(logger logr.Logger, namespace string) error {
+	ocdList := &operatorv2.OperatorConditionList{}
+	err := r.Client.List(context.TODO(), ocdList, client.InNamespace(namespace))
+	if err != nil {
+		logger.Error(err, "failed to list OperatorConditions")
+		return err
+	}
+
+	var condNotUpgradeable *metav1.Condition
+	var ocdName string
+	for ocdIdx := range ocdList.Items {
```

Comment:
Lets make use of starting csv names while comparing the operator conditions and ignore others.

Reply:
done.